js/reference.js

```diff
@@ -2716,18 +2724,60 @@
             }
             // heuristics
             else {
+                // fetch config option for system columns heuristics (true|false|Array)
+                // if true, add all system columns
+                // if false, add none
+                // if array, add the ones defined
+
+                // order of system columns will always be the same
+                // RID will always be first in the visible columns list
+                // the rest will always be at the end ('RCB', 'RMB', 'RCT', 'RMT')
+
+                var systemColumnsMode = null;
+                // if compact or detailed, check for system column config option
+                if (module._isCompactContext(this._context)) {
+                    systemColumnsMode = module._systemColumnsHeuristicsMode("compact");
+                } else if (this._context == module._contexts.DETAILED) {
+                    systemColumnsMode = module._systemColumnsHeuristicsMode("detailed");
+                }
+
+                // if array or true, add RID to the list of columns
+                if (systemColumnsMode) {
+                    if ((typeof systemColumnsMode == "object" && systemColumnsMode.indexOf("RID")) || systemColumnsMode == true) {
```

Comment:
What is this `(typeof systemColumnsMode == "object" && systemColumnsMode.indexOf("RID"))` condition checking for? This translates to "add RID display key if it's not the first element of the array" which I don't think that's what you want.

It's better to change the object check to actual array check (Array.isArray). Can RID be part of `systemColumnsMode`?

Reply:
The intention of this check was to check if it's an array (array's type is "object") and that RID is present in the array (that check should be `indexOf() != -1`). If that check fails, it verifies if the value is set to `true` instead.

`Array.isArray` is a better way to check this. If `RID` is not in the array of system columns, then it won't be added.